relay-pii/src/selector.rs

```diff
@@ -170,6 +170,11 @@ impl SelectorSpec {
         handle_selector(
             SelectorParser::parse(Rule::RootSelector, s)
                 .map_err(|e| InvalidSelectorError::ParseError(Box::new(e)))?
+                // Unwrapping here is guaranteed to work based on the grammar that was just
+                // successfully parsed.
+                //
+                // Extracts the first `OrSelector` spanning the entire selector from the `RootSelector`.
+                // The `RootSelector` is guaranteed to have exactly one `OrSelector`.
```

Comment:
The problem with this `unwraps` that is something changes in the future and we lose these guarantees, we can forget revisit this piece of code and behave differently.  What do you think? 

We prefer not to have the unwraps in the first place. 

Will it be possible to actually map them to error at the end (even if that can never occur now) and just use `?` to propagate it?

Reply:
I was thinking, these unwraps will never actually throw and our tests would verify this. 

But thinking about it, might as well just "yeet" some internal error and create a sentry issue, this will still be caught by tests (hopefully) and if it doesn't we at least get the sentry error.